x11-misc/dunst: Update USE flags

fedang opened the pull request:
Newer version of dunst support conditional compilation of certain features, like the X backend.

dunst-1.11.0-r1.ebuild and dunst-9999.ebuild were modified to reflect that.

NOTE: I made a revision because the RDEPEND changed

<!-- Please put the pull request description above -->

---

Please check all the boxes that apply:

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

ConiKost commented on x11-misc/dunst/dunst-9999.ebuild:
```diff
@@ -12,21 +12,25 @@ HOMEPAGE="https://dunst-project.org/ https://github.com/dunst-project/dunst"
 LICENSE="BSD"
 SLOT="0"
-IUSE="wayland"
+IUSE="+X wayland +dunstify +completions +xdg"
```
Please sort alphabetically.

ConiKost commented on x11-misc/dunst/dunst-1.11.0-r1.ebuild:
```diff
@@ -0,0 +1,90 @@
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~arm64 ~ppc ~ppc64 ~riscv ~x86"
+IUSE="+X wayland +dunstify +completions +xdg"
```
Please sort alphabetically.

ConiKost commented on x11-misc/dunst/dunst-9999.ebuild:
```diff
@@ -70,6 +79,9 @@ src_install() {
 		WAYLAND="$(usex wayland 1 0)"
+		X11="$(usex X 1 0)"
+		DUNSTIFY="$(usex dunstify 1 0)"
+		COMPLETIONS="$(usex completions 1 0)"
```
Please sort here also alphabetically.

ConiKost commented on x11-misc/dunst/dunst-9999.ebuild:
```diff
@@ -12,21 +12,25 @@ HOMEPAGE="https://dunst-project.org/ https://github.com/dunst-project/dunst"
 	dev-libs/glib:2
 	sys-apps/dbus
-	x11-libs/cairo[X,glib]
+	x11-libs/cairo[glib]
```
Instead adding ```x11-libs/cairo[X]``` to ```X? (``` you could do instead: ```x11-libs/cairo[X?,glib]```, so you don't have to add twice the same dep with different use flags.

fedang replied:
thanks, didn't know about that :+1:

ConiKost commented on x11-misc/dunst/dunst-1.11.0-r1.ebuild:
```diff
@@ -0,0 +1,90 @@
+		WAYLAND="$(usex wayland 1 0)"
+		X11="$(usex X 1 0)"
+		DUNSTIFY="$(usex dunstify 1 0)"
+		COMPLETIONS="$(usex completions 1 0)"
```
Please also sort here alphabetically.

ConiKost commented on x11-misc/dunst/dunst-9999.ebuild:
```diff
@@ -12,21 +12,25 @@ HOMEPAGE="https://dunst-project.org/ https://github.com/dunst-project/dunst"
-	x11-libs/libXinerama
-	x11-libs/libXrandr
-	x11-libs/libnotify
-	x11-libs/pango[X]
```
Same as obove for cairo.

ConiKost commented on x11-misc/dunst/dunst-1.11.0-r1.ebuild:
```diff
@@ -0,0 +1,90 @@
+	sys-apps/dbus
+	x11-libs/cairo[glib]
+	x11-libs/gdk-pixbuf:2
+	x11-libs/pango
```
Same as obove for cairo.

ConiKost commented on x11-misc/dunst/dunst-1.11.0-r1.ebuild:
```diff
@@ -0,0 +1,90 @@
+DEPEND="
+	dev-libs/glib:2
+	sys-apps/dbus
+	x11-libs/cairo[glib]
```
Instead adding ```x11-libs/cairo[X]``` to ```X? (``` you could do instead: ```x11-libs/cairo[X?,glib]```, so you don't have to add twice the same dep with different use flags.

ConiKost reviewed (changes requested):
Left some feedback :-)

fedang commented:
> Left some feedback :-)

Thank you, this was my first ebuild.
Is it okay now?

ConiKost commented on x11-misc/dunst/metadata.xml:
```diff
@@ -5,6 +5,11 @@
 	</maintainer>
+	<use>
+		<flag name="dunstify">Build dunstify (notify-send alternative)</flag>
+		<flag name="completions">Install shell completions (for bash, zsh and fish)</flag>
```
Also here please alphabetically.

fedang commented:
done :+1:

ConiKost commented on x11-misc/dunst/dunst-9999.ebuild:
```diff
@@ -12,21 +12,23 @@ HOMEPAGE="https://dunst-project.org/ https://github.com/dunst-project/dunst"
+		x11-libs/libXinerama
+		x11-libs/libXrandr
+	)
+	dunstify? ( x11-libs/libnotify )
```
Please look again through, this is also not sorted.

fedang replied:
ok done it.
is there an ebuild style guide so I don't have to make such mistakes?

fedang commented:
I have reordered the depends.

Also I wanted to mention this. Now dunst does not required X anymore.
Since it equally supports wayland and x11, maybe it would be more appropriate to be in app-misc or gui-apps (like mako), rather than x11-misc?
But this is just a suggestion, I don't know the policy on changing categories

ConiKost commented on x11-misc/dunst/metadata.xml:
```diff
@@ -5,6 +5,11 @@
 		<name>Conrad Kostecki</name>
 	</maintainer>
+	<use>
+		<flag name="completions">Install shell completions (for bash, zsh and fish)</flag>
```
One last small nit pick. Please order ```bash, fish and zsh```. Thanks! Otherwise lgtm now.

fedang replied:
done :+1:

ConiKost commented:
Yes, I think its worth to add it. You can also drop 1.11.0-r1 commits, given that there is now a new release, I would tend to add you changes about use flags etc. only to newest 1.12.0 version.

fedang commented:
I removed 1.11.0-r1.

Also added in `pkg_postinst` a message about the breaking change.
I did not include the full message for length. But if you prefer the entire thing let me know.

ConiKost commented on x11-misc/dunst/dunst-1.12.0.ebuild:
```diff
@@ -0,0 +1,99 @@
+}
+
+pkg_postinst() {
+	einfo "The behaviour of the setting 'height' has been changed in a breaking way."
```
While the text is fine, this should be only display for users, who are upgrading. See, for example, ```net-im/prosody```, how you could do that.

fedang replied:
👍

fedang commented:
@ConiKost I just added a check for the REPLACING_VERSIONS.
I think patching the user config is not advisable

ConiKost reviewed (changes requested):
thanks! lgtm. But please squash your commits.

fedang commented:
> thanks! lgtm. But please squash your commits.

I squashed to a single commit
